word_segmentation.py

```diff
@@ -205,140 +294,115 @@ def clean_line(line):
-                        if line[i] == '|':
-                            word_brkpoints.append(i - found_bars)
-                            found_bars += 1
-
-                    # Creating the unsegmented line
-                    unsegmented_line = line.replace("|", "")
-
-                    # Making the grapheme clusters brkpoints
-                    chars_break_iterator.setText(unsegmented_line)
-                    char_brkpoints = [0]
-                    for brkpoint in chars_break_iterator:
-                        char_brkpoints.append(brkpoint)
+                    line = Line(file_line, "man_segmented")
 
                     # Storing the grapheme clusters and their frequency in the dictionary
-                    for i in range(len(char_brkpoints) - 1):
-                        grapheme_clusters_dic[
-                            unsegmented_line[char_brkpoints[i]: char_brkpoints[i + 1]]] = grapheme_clusters_dic.get(
-                            unsegmented_line[char_brkpoints[i]: char_brkpoints[i + 1]], 0) + 1
-                    true_bies = get_bies(char_brkpoints, word_brkpoints)
-                    true_bies_str = get_bies_string_from_softmax(np.transpose(true_bies))
+                    for i in range(len(line.char_brkpoints) - 1):
+                        new_graph_clust = line.unsegmented[line.char_brkpoints[i]: line.char_brkpoints[i + 1]]
+                        grapheme_clusters_dic[new_graph_clust] += 1
 
-                    # Compute segmentations of icu and BIES associated with it
-                    words_break_iterator.setText(unsegmented_line)
-                    icu_word_brkpoints = [0]
-                    for brkpoint in words_break_iterator:
-                        icu_word_brkpoints.append(brkpoint)
-                    icu_word_segmented_str = get_segmented_string(unsegmented_line, icu_word_brkpoints)
-                    icu_bies = get_bies(char_brkpoints, icu_word_brkpoints)
+                    # Computing BIES corresponding to the manually segmented and ICU segmented
+                    true_bies = get_bies(line.char_brkpoints, line.man_word_brkpoints)
+                    true_bies_str = get_bies_string_from_softmax(np.transpose(true_bies))
+                    icu_bies = get_bies(line.char_brkpoints, line.icu_word_brkpoints)
                     icu_bies_str = get_bies_string_from_softmax(np.transpose(icu_bies))
 
                     # Counting the number of mismatches between icu_bies and true_bies
                     icu_total_bies_lengths += len(icu_bies_str)
                     icu_mismatch += diff_strings(true_bies_str, icu_bies_str)
 
+                    # Computing the F1-score between icu_bies and true_bies
+                    f1_output = compute_f1_score(true_bies=true_bies_str, est_bies=icu_bies_str)
+                    correctly_segmented_words += f1_output[1]
+                    true_words += f1_output[2]
+                    segmented_words += f1_output[3]
```

Comment:
If the F1 score is a structured output, you should return a type like a NamedTuple